app/scripts/controllers/rif/rns/resolver/index.js

```diff
@@ -204,16 +204,29 @@ export default class RnsResolver extends RnsJsDelegate {
    * @param chain ChainId that you want to delete from the pending queue
    * @returns {Promise<unknown>}
    */
-  deletePendingChainAddress (chain, isSubdomain) {
+  deletePendingChainAddress (domainName, chain, isSubdomain) {
     return new Promise((resolve, reject) => {
-      const index = this.store.pendingChainAddressesActions.findIndex((e) => e.chain === chain && e.isSubdomain === isSubdomain);
+      const domain = this.getDomainForPendingChainAddresses(domainName);
+      const pendingChainAddressesActions = domain.pendingChainAddresses;
+      const index = pendingChainAddressesActions.findIndex((e) => e.chain === chain && e.isSubdomain === isSubdomain);
       if (index >= 0) {
-        this.store.pendingChainAddressesActions.splice(index, 1);
+        pendingChainAddressesActions.splice(index, 1);
       }
+      this.updateDomain(domain);
       resolve();
     });
   }
 
+  getDomainForPendingChainAddresses (domainName) {
```

Comment:
I think this method is unnecessary, you could do that check inside the current getDomain method, also updating a domain inside a method which the name is get something is not right, although is not necessary to update that domain since the array is empty each time you call the getDomain method, you should do the check and set an empty array there. Think that this case is only for those domains that you already have in the store so this is only going to happen to you on that particular case, i don't think this is necessary.

Reply:
This code is deleted in next commit